src/braket/pulse/pulse_sequence.py

```diff
@@ -311,6 +311,11 @@ def to_ir(self) -> str:
             str: a str representing the OpenPulse program encoding the PulseSequence.
         """
         program = deepcopy(self._program)
+        program.autodeclare(encal=False)
+        sorted_parameters = sorted(self.parameters, key=lambda p: p.name, reverse=True)
```

Comment:
Naive question but does this have to be sorted? Curious as to why.

Reply:
If ordering doesn't matter, we can do a check of the parameter strings using set comparison between the split multi-line strings.
```
a = set(string.split(string_a))
b = set(string.split(string_a))
assert a == b
```
I think this approach creates some lengthier tests. The sorting slow down should not be too bad here.